internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -25,54 +26,127 @@ const (
 
 const (
 	annotationCPULoadBalancing = "cpu-load-balancing.crio.io"
+	annotationCPUQuota         = "cpu-quota.crio.io"
+	annotationIRQLoadBalancing = "irq-load-balancing.crio.io"
 	schedDomainDir             = "/proc/sys/kernel/sched_domain"
+	irqSmpAffinityProcFile     = "/proc/irq/default_smp_affinity"
+	cgroupMountPoint           = "/sys/fs/cgroup"
 )
 
 // HighPerformanceHooks used to run additional hooks that will configure a system for the latency sensitive workloads
 type HighPerformanceHooks struct{}
 
 func (h *HighPerformanceHooks) PreStart(ctx context.Context, c *oci.Container, s *sandbox.Sandbox) error {
 	log.Infof(ctx, "Run %q runtime handler pre-start hook for the container %q", HighPerformance, c.ID())
+
+	if isCgroupParentBurstable(s) {
+		log.Infof(ctx, "Container %q is a burstable pod. Skip PreStart.", c.ID())
+		return nil
+	}
+	if isCgroupParentBestEffort(s) {
+		log.Infof(ctx, "Container %q is a besteffort pod. Skip PreStart.", c.ID())
+		return nil
+	}
+	if !isContainerRequestWholeCPU(c) {
+		log.Infof(ctx, "Container %q requests partial cpu(s). Skip PreStart", c.ID())
+		return nil
+	}
+
 	// disable the CPU load balancing for the container CPUs
 	if shouldCPULoadBalancingBeDisabled(s.Annotations()) {
+		log.Infof(ctx, "Disable cpu load balancing for container %q", c.ID())
 		if err := setCPUSLoadBalancing(c, false, schedDomainDir); err != nil {
 			return err
 		}
 	}
+	// disable the IRQ smp load balancing for the container CPUs
+	if shouldIRQLoadBalancingBeDisabled(s.Annotations()) {
+		log.Infof(ctx, "Disable irq smp balancing for container %q", c.ID())
+		if err := setIRQLoadBalancing(c, false, irqSmpAffinityProcFile); err != nil {
+			return err
+		}
+	}
+	// disable the CFS quota for the container CPUs
+	if shouldCPUQuotaBeDisabled(s.Annotations()) {
+		log.Infof(ctx, "Disable cpu cfs quota for container %q", c.ID())
+		cpuMountPoint, err := cgroups.FindCgroupMountpoint(cgroupMountPoint, "cpu")
+		if err != nil {
+			return err
+		}
+		if err := setCPUQuota(cpuMountPoint, s.CgroupParent(), c, false); err != nil {
+			return err
+		}
+	}
 
 	return nil
 }
 
 func (h *HighPerformanceHooks) PreStop(ctx context.Context, c *oci.Container, s *sandbox.Sandbox) error {
 	log.Infof(ctx, "Run %q runtime handler pre-stop hook for the container %q", HighPerformance, c.ID())
+
+	if isCgroupParentBurstable(s) {
+		log.Infof(ctx, "Container %q is a burstable pod. Skip PreStart.", c.ID())
```

Comment:
Should these log messages in `PreStop()` say "PreStop"?

```suggestion
		log.Infof(ctx, "Container %q is a burstable pod. Skip PreStop.", c.ID())
```

Reply:
good catch